engine-precompiles/src/alt_bn256.rs

```diff
@@ -67,39 +159,52 @@ fn read_point(input: &[u8], pos: usize) -> Result<bn::G1, ExitError> {
     })
 }
 
-pub(super) struct Bn128Add<HF: HardFork>(PhantomData<HF>);
+pub(super) struct Bn256Add<HF: HardFork>(PhantomData<HF>);
 
-impl<HF: HardFork> Bn128Add<HF> {
+impl<HF: HardFork> Bn256Add<HF> {
     pub(super) const ADDRESS: Address = super::make_address(0, 6);
 
     pub fn new() -> Self {
         Self(Default::default())
     }
 }
 
-impl<HF: HardFork> Bn128Add<HF> {
+impl<HF: HardFork> Bn256Add<HF> {
     fn run_inner(input: &[u8], _context: &Context) -> Result<Vec<u8>, ExitError> {
-        use bn::AffineG1;
-
         let mut input = input.to_vec();
         input.resize(consts::ADD_INPUT_LEN, 0);
 
         let p1 = read_point(&input, 0)?;
-        let p2 = read_point(&input, 64)?;
-
-        let mut output = [0u8; 64];
-        if let Some(sum) = AffineG1::from_jacobian(p1 + p2) {
-            let x = sum.x().into_u256().to_big_endian();
-            let y = sum.y().into_u256().to_big_endian();
-            output[0..32].copy_from_slice(&x);
-            output[32..64].copy_from_slice(&y);
-        }
+        let p2 = read_point(&input, consts::POINT_LEN)?;
 
+        let output = Self::execute(p1, p2)?;
         Ok(output.to_vec())
     }
+
+    #[cfg(not(feature = "contract"))]
+    fn execute(p1: bn::G1, p2: bn::G1) -> Result<[u8; consts::OUTPUT_LEN], ExitError> {
+        let mut output = [0u8; consts::POINT_LEN];
+        if let Some(sum) = bn::AffineG1::from_jacobian(p1 + p2) {
+            sum.x()
+                .to_big_endian(&mut output[0..consts::SCALAR_LEN])
+                .map_err(|_e| ExitError::Other(Borrowed("ERR_BIG_ENDIAN")))?;
```

Comment:
Could factor out `ExitError::Other(Borrowed("ERR_BIG_ENDIAN"))` as a constant since it appears multiple times throughout these precompiles.

Reply:
Evgeny and I found out that WASM magically only states them once if it notices that there are multiple. Regardless, for sake of keeping up code quality, will do so.